File: sixth/EsPrimo.py
'''
Crea un programa que pida al usuario un número
e indique al usuario si es primo o no.
 
 @author: Gonzalo Pulleiro
 
'''

import math

num = int(input("Introduce un número...\n"))

esPrimo = True

if(num <= 1):
    esPrimo = False
else:
    for x in range(2, int(math.sqrt(num)) + 1):
        if(num % x == 0):
            esPrimo = False
            break


if(not esPrimo):
    print("{} no es un número primo.".format(num))
else:
    print("{} es un número primo.".format(num))

File: Sencillos/sixth/EsPrimo.java
package sixth;

/*
 * Crea un programa que pida al usuario un número
 * e indique al usuario si es primo o no.
 * 
 *  @author: Gonzalo Pulleiro
 */

import java.util.Scanner;

public class EsPrimo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Introduce un número...");
        int num = sc.nextInt();
        sc.close();

        boolean esPrimo = true;

        if (num <= 1) {
            esPrimo = false;
        } else {
            for (int i = 2; i <= Math.sqrt(num); i++) {
                if (num % i == 0) {
                    esPrimo = false;
                    break;
                }
            }
        }

        if (esPrimo) {
            System.out.println(num + " es un número primo.");
        } else {
            System.out.println(num + " no es un número primo");
        }
    }
}
